File: java/java-sdk/src/main/java/io/kestra/sdk/api/QuotasApi.java
package io.kestra.sdk.api;

import com.fasterxml.jackson.core.type.TypeReference;

import io.kestra.sdk.internal.ApiClient;
import io.kestra.sdk.internal.ApiException;
import io.kestra.sdk.internal.BaseApi;
import io.kestra.sdk.internal.Configuration;

import io.kestra.sdk.model.QuotaLimit;
import io.kestra.sdk.model.QuotaLimitControllerApiQuotaLimitResetRequest;

import java.util.List;

public class QuotasApi extends BaseApi {

    public QuotasApi() {
        super(Configuration.getDefaultApiClient());
    }

    public QuotasApi(ApiClient apiClient) {
        super(apiClient);
    }

    // ========================================================================
    // Quota limits
    // ========================================================================

    public List<QuotaLimit> searchQuotaLimits(
            @jakarta.annotation.Nonnull String tenant) throws ApiException {
        return invoke("GET",
                tenantPath(tenant, "quota-limits"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public void resetQuotaLimit(
            @jakarta.annotation.Nonnull String tenant,
            @jakarta.annotation.Nonnull QuotaLimitControllerApiQuotaLimitResetRequest request) throws ApiException {
        invoke("POST",
                tenantPath(tenant, "quota-limits", "reset"),
                request, null, null,
                JSON, JSON, null);
    }

}

File: python/python-sdk/kestrapy/api/quotas_api.py
from typing import Any, List

from kestrapy.base_api import BaseApi
from kestrapy.models.quota_limit import QuotaLimit
from kestrapy.models.quota_limit_controller_api_quota_limit_reset_request import QuotaLimitControllerApiQuotaLimitResetRequest


class QuotasApi(BaseApi):

    def search_quota_limits(self, tenant: str) -> List[QuotaLimit]:
        path = self._tenant_path(tenant, "quota-limits")
        return self._json_list_request("GET", path, QuotaLimit)

    def reset_quota_limit(self, tenant: str, request: QuotaLimitControllerApiQuotaLimitResetRequest) -> Any:
        path = self._tenant_path(tenant, "quota-limits", "reset")
        return self._raw_json_request("POST", path, body=request)
